server/uploader.js

```diff
@@ -95,41 +99,47 @@ uploader.bigWigToBCWig = function(prefix, bwFile, bigWigToWigAddr) {
       if (!seg.hasOwnProperty(linePart[0])) {
         seg[linePart[0]] = [];
       }
-      if (xl != lastxr && lastxr > -1) {
+      if (xl != lastChrXr[chName] && lastChrXr[chName] > -1) {
         seg[chName].push({
-          x: lastxr,
-          val: 0
+          x: lastChrXr[chName],
+          value: 0.0
         });
       }
       seg[chName].push({
         x: xl,
-        val: val
+        value: val
       });
-      lastxr = xr;
+      lastChrXr[chName] = xr;
+      lastValue[chName] = val;
     }
   });
 
   lines.on('close', function() {
+    for (var chr in lastChrXr) {
+      seg[chr].push({
+        x: lastChrXr[chr],
+        value: lastValue[chr]
+      });
+    }
     // write to *.bcwig file
     // console.log('start log it');
     // if the folder already exists, then delete it
     var folder = prefix + bwFile + '_chr';
-    var stats = fs.lstatSync(folder);
-    if (stats.isDirectory()) {
+    if (fs.existsSync(folder)) {
       fs.rmdirSync(folder);
       console.log('Wiggle file ' + bwFile + ' is replaced.');
     }
-    mkdirp(folder);
+    fs.mkdirSync(folder);
 
     for (var chr in seg) {
       var bcwigFile = folder + '/' + bwFile + '_' + chr + '.bcwig';
-      var fd = fs.openSync(bcwigFile, 'w');
+      var bcwigBuf = new Buffer(12 * seg[chr].length);
```

Comment:
Seem to be too many constants around. Please create variables for 12 and explain in comments its meaning.